build.gradle

```diff
@@ -0,0 +1,149 @@
+buildscript {
+    repositories {
+        gradlePluginPortal()
+    }
+}
+
+plugins {
+    id "com.palantir.docker" version "0.25.0"
+    id "com.palantir.docker-run" version "0.25.0"
+}
+
+// needed for palantir plugin
+task build {
+}
+
+// docker & docker run
+docker {
+    name "unit8/darts"
+    tags version, "latest"
+    dockerfile file("${project.rootDir}/Dockerfile")
+    // needed files for docker and to build library
+    files "README.md", "setup.py", "setup.cfg"
+    copySpec.with {
+        from(".") {
+            include "examples/**"
+            into "."
+        }
+        from(".") {
+            include "darts/**"
+            into "."
+        }
+        from(".") {
+            include "requirements/**"
+            into "."
+        }
+    }
+}
+
+dockerRun {
+    name "unit8_darts"
+    image "unit8/darts:latest"
+    ports "8888:8888"
+    daemonize false
+    clean true
+}
+
+// setup requirements
+task setupPip(type: Exec) {
+    commandLine "python", "-m", "pip", "install", "--upgrade", "pip"
+}
+
+void pipInstallSteps() {
```

Comment:
how can we pick some of those steps? some build tasks need main + release, some need main + dev so installing all of those is not needed

Reply:
In that form we couldn't but now it is splitted into 3 task and in other places tasks now depend only on needed requirements.